NEWS

```diff
@@ -1,231 +1,444 @@
+
+    - hfile_s3 now uses version 4 signatures by default.  Attempting to write to
+      an S3 bucket will also now work correctly.  It is possible to force
+      version 2 signatures by creating environment variable HTS_S3_V2 (the exact
+      value does not matter, it just has to exist).  Note that writing depends
+      on features that need version 4 signatures, so forcing version 2 will
+      disable writes.
+
+    - hfile_s3 will automatically retry requests where the region endpoint
+      was not specified correctly, either by following the 301 redirect (when
+      using path-style requests) or reading the 400 response (when using
+      virtual-hosted style requests and version 4 signatures).  The first
+      region to try can be set by using the AWS_DEFAULT_REGION environment
+      variable, by setting "region" in ".aws/credentials" or by setting
+      "bucket_location" in ".s3cfg".
+
+    - hfile_s3 now percent-escapes the path component of s3:// URLs.  For
+      backwards-compatibility it will ignore any paths that have already
+      been escaped (detected by looking for '%' followed by two hexadecimal
+      digits.)
+
+    - New environment variables HTS_S3_V2, HTS_S3_HOST, HTS_S3_S3CFG
+      and HTS_S3_PART_SIZE to force version-2 signatures, control the
+      S3 server hostname, the configuration file and upload chunk
+      sizes respectively.
+
+ * Numerous SAM format improvements.
+
+   - Bgzipped SAM files can now be indexed and queried.  The library now
+     recognises sam.gz as a format name to ease this usage.
+
+   - The SAM reader and writer now supports multi-threading via the
+     thread-pool.
+
+     Note that the multi-threaded SAM reader does not currently support seek
+     operations.  Trying to do this (for example with an iterator range request)
+     will result in the SAM readers dropping back to single-threaded mode.
+
+   - Major speed up of SAM decoding and encoding, by around 2x.
+
+   - SAM format can now handle 64-bit coordinates and references.  This
+     has implications for the ABI too (see below).  Note BAM and CRAM
+     currently cannot handle references longer than 2Gb, however given
+     the speed and threading improvements SAM.gz is a viable workaround.
+
+ * We can now automatically build indices on-the-fly while writing
+   SAM, BAM, CRAM, VCF and BCF files.  (Note for SAM and VCF this only
+   works when bgzipped.)
+
+ * HTSlib now supports the @SQ-AN header field, which lists alternative names
+   for reference sequences.  This means given "@SQ SN:1 AN:chr1", tools like
+   samtools can accept requests for "1" or "chr1" equivalently.  (PR #931)
+
+ * Completely blank files are no longer considered to be a SAM file
+   with no header and no alignments.  Thus pipelines such as `somecmd
+   | samtools ...` with `somecmd` aborting before outputting anything
+   will not propagate the error to the second command. (#261)
```

Comment:
s/not/now/!!

But I don't really see why you've rewritten the paragraph entirely. Previous was

```
* Empty (0-length) files are no longer considered to be valid SAM files.
  Input files without any headers or records at all are usually seen in
  pipelines such as `somecmd | samtools ...` with `somecmd` aborting and
  outputting nothing, and it is beneficial for the second command to
  propagate the error.  (#261)
```

This wording was carefully chosen. “Empty (0-length)” is clearer and more precise than “Completely blank”; “blank files … a SAM file” is something of a number mismatch; the user documentation uses both “records” and “alignments” and in this context I think the former is clearer; the line break around the pipe was avoided; the thing that is doing the propagating is the second command itself not some other magic. OTOH _aborting before outputting anything_ is an improvement and “it is beneficial for” was awkward and would benefit from some rewording.

Reply:
The reason I rewrote it is mainly because trying to merge changes listed in around 5000 lines of git logs with a couple thousand lines of sporadic NEWs comments, which are all reordered, was just nigh on impossible.

Instead I started again from scratch, and then searched through the old news file to see what was worth lifting over and to check I hadn't missed anything.  I found plenty of cases where I'd rephrased it, but generally prefer brevity over excessive verbiage.  People can look at the git logs if they want the fine detail.  We're not trying to write a specification.  Words like "blank" are fine IMO if it saves extra waffle.

Reply:
I appreciate that this was a lot of effort; reviewing it is going to be quite an effort too. Should I assume that the indentation nits will be fixed up before I proofread the whole thing again, or proofread it as is?

Brevity is good of course, as is accuracy and clarity. Where effort has been made over the last 16 months to write clear accurate NEWS entries contemporaneously with the associated changes, it seems unfortunate to discard them.

Reply:
Minor formatting problems will be fixed before this is merged.  Unfortunately it might take a bit longer than expected to get on to this as my equivalent updates to samtools just got blown away by a brief power cut. :disappointed: